src/libraries/System.Private.CoreLib/src/System/DateTime.cs

```diff
@@ -1349,32 +1342,28 @@ public long ToBinary()
         // corresponds to this DateTime with the time-of-day part set to
         // zero (midnight).
         //
-        public DateTime Date
-        {
-            get
-            {
-                ulong ticks = UTicks;
-                return new DateTime((ticks - ticks % TimeSpan.TicksPerDay) | InternalKind);
-            }
-        }
+        public DateTime Date => GetDate(_dateData);
+        private static DateTime GetDate(ulong data) => new DateTime(((data & TicksMask) / TimeSpan.TicksPerDay * TimeSpan.TicksPerDay) | (data & FlagsMask));
```

Comment:
As the JIT evolving, this is likely to have less impact than it did in #58169, which should already be trivial then.